core/src/main/java/io/bisq/core/btc/wallet/WalletsSetup.java

```diff
@@ -300,35 +300,41 @@ private void configPeerNodesForLocalHostBitcoinNode() {
 
     private void configPeerNodes(Socks5Proxy socks5Proxy) {
         List<BitcoinNodes.BtcNode> btcNodeList = new ArrayList<>();
-        walletConfig.setMinBroadcastConnections((int) Math.floor(DEFAULT_CONNECTIONS * 0.8));
         switch (BitcoinNodes.BitcoinNodesOption.values()[preferences.getBitcoinNodesOptionOrdinal()]) {
             case CUSTOM:
-                String bitcoinNodesString = preferences.getBitcoinNodes();
-                if (bitcoinNodesString != null) {
+                final String bitcoinNodesString = preferences.getBitcoinNodes();
+                if (bitcoinNodesString != null && !bitcoinNodesString.isEmpty()) {
                     btcNodeList = Splitter.on(",")
                             .splitToList(StringUtils.deleteWhitespace(bitcoinNodesString))
                             .stream()
                             .filter(e -> !e.isEmpty())
                             .map(BitcoinNodes.BtcNode::fromFullAddress)
                             .collect(Collectors.toList());
+                    walletConfig.setMinBroadcastConnections((int) Math.ceil(btcNodeList.size() * 0.5));
+                    // If Tor is set we usually only use onion nodes, but if user provides mixed clear net and onion nodes we want to use both
+                    useAllProvidedNodes = true;
+                } else {
+                    log.warn("Custom nodes is set but no nodes are provided. We fall back to provided nodes option.");
+                    preferences.setBitcoinNodesOptionOrdinal(BitcoinNodes.BitcoinNodesOption.PROVIDED.ordinal());
+                    btcNodeList = bitcoinNodes.getProvidedBtcNodes();
+                    walletConfig.setMinBroadcastConnections(4);
                 }
                 break;
```

Comment:
The `else` clause says we're going to "fall back" to the provided nodes option, and it then duplicates the code in the actual `PROVIDED` case block below. Why not remove the `break` statement here, move the `PUBLIC` case above this case, and let the logic here _actually_ fall though to the `PROVIDED` case below?

You can keep the `log.warn` statement, and `setBitcoinNodesOptionOrdinal` call, but the duplicate assignment of `btcNodeList` and call to `walletConfig.setMinBroadcastConnections` could be eliminated.